### DISCUSSION TITLE
Ansible task failure during adduser

### DISCUSSION THREAD

**Discussion by jamesridley** [leighmacdonald/uncletopia#134] [Q&A] [answered]
Hi there, I'm trying to get this running on a VM and I'm running into a bit of trouble. I've got two VMs set up with one acting as the local host and the other acting as a server. I've got the SSH keys for connecting from the local host to the server root set up, but when I run `make adduser` on the local host I get the following error:  (I've omitted the server IP just in case)

```
TASK [adduser : ssh keys] **************************************************************************************************
fatal: [<serverIP>]: FAILED! => {"msg": "'ssh_keys' is undefined"}

PLAY RECAP *****************************************************************************************************************
<serverIP>            : ok=13   changed=2    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

make: *** [Makefile:9: adduser] Error 2
```

I'm not sure what's wrong here. It connects normally and completes every task up until this one. Would appreciate some help as I'm trying to implement a Uncletopia style server for my local TF2 scene. :)

**Comment by jamesridley** [comment]
Okay, the problem was that I didn't set up my all.yml file. But now I have a similair issue with `map_rotation`: when deploying it fails on the `mapcycle.txt` task. 

`'map_rotation' is undefined` 

I'm struggling to understand what is missing... I have a single map placed in `tf/maps` and have updated `mapcycle.txt` accordingly. I fiddled with a few things, but I can't get the `map_rotation` to work. I'm sure it's a very simple fix but rn I'm stuck. Also I forgot to thank you for making this project open-source in my original post, so, thank you!

**Comment by leighmacdonald** [comment, marked as answer]
This is probably because the example configs are a bit out of date and missing some items, I have just updated them. Let me know if you still have issues.

You can see the changes in https://github.com/leighmacdonald/uncletopia/commit/b172249f7f245e32ad9433bc9e01994698e8894a
